dCommon/AMFDeserialize.cpp

```diff
@@ -0,0 +1,151 @@
+#include "AMFDeserialize.h"
+
+#include "AMFFormat.h"
+
+/**
+ * AMF3 Reference document https://rtmp.veriskope.com/pdf/amf3-file-format-spec.pdf
+ * I have only written the values that the LEGO Universe client sends to the server.  All others are left out.
+ * AMF3 Deserializer written by EmosewaMC
+ */
+AMFArrayValue* AMFDeserialize::Read(RakNet::BitStream* inStream, bool readStartOfArray, std::string parentKey) {
```

Comment:
interesting.  Ill look at making the method static and also enhancing the deletion process of amf arrays.  currently they dont have any destructors at all which could cause leaks since those values may be forgotten about.  Ill look at the unique_ptr as well.  Not something Im too familiar with but ill give it a go regardless

Reply:
AMF and LDF should both ideally utilize smart pointers, as deletion is seldom consistent and **does** cause sizable memory leaks. Smart pointers aren't used anywhere else in the codebase, so would require quite a bit of refactoring to introduce them. Would suggest a separate PR for that though.

Reply:
Interesting.  Since this code isnt going to be used in the live game until I implement the case in the message handler, ill possibly put that in before I put in the actual property behavior stuff to continue cleaning up the memory since any user could create lots of memory leaks very quickly with large AMFs the client can send.

Reply:
Addressed this for the tests - unique_ptr re-write for AMF will need to wait for another PR.